src/sagas/account.js

```diff
@@ -27,32 +28,40 @@ import { getTokenName } from '../selectors/env';
 import { minDelayCall } from './helpers';
 
 export function * doSignup (action) {
-  const { formData, nextRoute } = action;
+  const { formData } = action;
 
   try {
     const response = yield minDelayCall(api.signup, formData);
 
+    if (_.has(formData, 'nextRoute.path')) {
+      _.assign(response, {nextRoute: formData.nextRoute});
+    }
+
     // Persist token for future page loads
     const tokenName = yield select(getTokenName);
-    new Cookies().set(tokenName, response.key);
+    new Cookies().set(tokenName, response.key, { path: '/' });
 
-    yield put({ type: SIGNUP_SUCCESS, response, nextRoute });
+    yield put({ type: SIGNUP_SUCCESS, response });
```

Comment:
This can should still use the obj explode operator so that `user` and `key` are accessible directly from the action payload object. This can also be updated to include `nextRoute` in the payload by doing:
``` yield put({ type: SIGNUP_SUCCESS, { ...response, nextRoute }}); ```

This should give you the following as your action payload: `{ user: <user>, key: <key>, nextRoute: <path or null>}`